tests/test_loss_mitigation.py

```diff
@@ -70,3 +75,64 @@ def test_photon_loss_mitigation(result_type):
     for keys, value in mitigated_dist.items():
         assert sum(keys) == ideal_photon_count
         assert value > 1e-6
+
+def _compute_random_circ_probs(source_emission, num_photons):
+
+    mzi = (BS() // (0, PS(phi=Parameter("φ_a")))
+           // BS() // (1, PS(phi=Parameter("φ_b"))))
+
+    random_loc = Circuit.decomposition(Matrix.random_unitary(20), mzi,
+                                              phase_shifter_fn=PS,
+                                              shape="triangle")
+    # Processor config
+    processor = Processor("SLOS", random_loc)
+    processor.min_detected_photons_filter(0)
+    processor.thresholded_output(True)
+
+    # Source properties
+    src = Source(emission_probability=source_emission)
+    processor.source = src
+
+    # Input state
+    input_state = BasicState([1] * num_photons + [0] * (random_loc.m - num_photons))
+
+    processor.with_input(input_state)
+
+    # Sampler
+    sampler = Sampler(processor)
+    return sampler.probs()['results']
+
+def total_variation_distance(p1, p2):
+    # calculate Total Variation distance between 2 probability distributions
+    tvd = 0.5 * np.sum(np.abs(np.array(p1) - np.array(p2)))
+    return tvd
+
+def test_mitigation_over_postselect_tvd():
+
+    ideal_photon_count = 4
+    # lossless distribution
+    ideal_dist = _compute_random_circ_probs(source_emission=1, num_photons=ideal_photon_count)
+    # lossy distribution
+    lossy_dist = _compute_random_circ_probs(source_emission=0.3, num_photons=ideal_photon_count)
+
+    # compute the migitated distribution from
+    mitigated_dist = photon_recycling(lossy_dist, ideal_photon_count)
+
+    # computing the postselected distribution for this random computation.
+    # Using the internal methods of loss_mitigation codes for this
+    # as the PostSelect function to set on Processor is not known
+
+    num_modes = next(iter(lossy_dist)).m  # number of modes
+    pattern_map = _generate_one_photon_per_mode_mapping(num_modes, ideal_photon_count)
+    noisy_distributions = _gen_lossy_dists(lossy_dist, ideal_photon_count, pattern_map)
+    post_dist = noisy_distributions[0]
+
+    for key, values in ideal_dist.items():
+        if key not in mitigated_dist.keys():
+            mitigated_dist[key] = 0.0
+
+    # computing tvd
+    tvd_miti = total_variation_distance(list(ideal_dist.values()), list(mitigated_dist.values()))
+    tvd_ps = total_variation_distance(list(ideal_dist.values()), list(post_dist))
+
+    assert tvd_miti , tvd_ps  # checks that mitigated is closer to ideal than post-selected distribution
```

Comment:
What does this assert do?
Doesn't it check that the tuple `tvd_miti , tvd_ps` (yes it's a tuple) is not empty. Which it cannot be, as it has 2 elements by construction.

Reply:
This assert is my typo, I entered ', ' instead of '<' check. A good catch :-)